pkg/certificate/interface.go

```diff
@@ -0,0 +1,101 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	Auto        ProviderType = "auto"
+	CertManager ProviderType = "cert-manager"
+	// add more ...
+)
+
+func NewProvider(pt ProviderType) Provider {
+	switch pt {
+	case Auto:
+		return nil // change me later
+	case CertManager:
+		return nil // change me later
+	}
+	panic(fmt.Sprintf("unknown provider type: %s", pt))
+}
+
+type Config struct {
+	CommonName       string
+	Organization     []string
+	DNSNames         []string
+	ValidityDuration time.Duration
+	CA               []byte
+
+	// ExtraConfig contains provider specific configurations.
+	ExtraConfig map[string]string
```

Comment:
could be `map[string]interface{}` for more flexibility and leaving it to the provider to cast.

Reply:
updated & thx